src/modules/Transactions/hooks/useSignTransaction.js

```diff
@@ -0,0 +1,97 @@
+/* eslint-disable max-statements */
+import React, { useEffect, useState } from 'react';
+
+import { useModal } from 'hooks/useModal';
+import ConfirmTransaction from '../components/ConfirmTransaction';
+import SignTransactionSuccess from '../components/SignTransaction/SignTransactionSuccess';
+import SignTransactionError from '../components/SignTransaction/SignTransactionError';
+
+export function useSignTransaction({
```

Comment:
```suggestion
export function useSignTransactionModal({
```
can we add the `Modal` suffix to be more clear?

Reply:
also rename the file to that name :)